Cannot Compile on Raspberry Pi 4

### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Steps to reproduce the issue

#### I first cloned Open3D version 0.16 release, Then, I build Open3D (on a fresh Raspbian 64-bit OS installed on Raspberry Pi 4 Model B with 8GB RAM) with:
```
mkdir build
cd build
cmake \
    	-DBUILD_GUI=OFF \
    	-DPYTHON_EXECUTABLE=$(which python3) \
    	-DBUILD_ISPC_MODULE=OFF \
    	-DGLIBCXX_USE_CXX11_ABI=ON \
    	-DCMAKE_CXX_FLAGS_RELEASE="-O3 -DNDEBUG -faligned-new" \
    	..
make -j3
```

### Error message

```shell
[ 20%] Generating ../common/scene_line_segments.cpp.avx2.cpp
[ 20%] Generating ../common/scene_points.cpp.avx2.cpp
[ 20%] Generating ../common/scene_quad_mesh.cpp.avx2.cpp
[ 21%] Generating ../common/scene_triangle_mesh.cpp.avx2.cpp
[ 22%] Generating ../common/scene_user_geometry.cpp.avx2.cpp
[ 22%] Generating ../geometry/curve_intersector_virtual_4i.cpp.avx2.cpp
[ 23%] Generating ../geometry/curve_intersector_virtual_4i_mb.cpp.avx2.cpp
[ 24%] Generating ../geometry/curve_intersector_virtual_4v.cpp.avx2.cpp
[ 24%] Generating ../geometry/curve_intersector_virtual_8i.cpp.avx2.cpp
[ 25%] Generating ../geometry/curve_intersector_virtual_8i_mb.cpp.avx2.cpp
[ 25%] Generating ../geometry/curve_intersector_virtual_8v.cpp.avx2.cpp
[ 25%] Building CXX object kernels/CMakeFiles/embree_avx2.dir/__/geometry/instance_intersector.cpp.avx2.cpp.o
c++: error: unrecognized command-line option ‘-mf16c’
c++: error: unrecognized command-line option ‘-mavx2’
c++: error: unrecognized command-line option ‘-mfma’
c++: error: unrecognized command-line option ‘-mlzcnt’
c++: error: unrecognized command-line option ‘-mbmi’
c++: error: unrecognized command-line option ‘-mbmi2’
make[5]: *** [kernels/CMakeFiles/embree_avx2.dir/build.make:220: kernels/CMakeFiles/embree_avx2.dir/__/geometry/instance_intersector.cpp.avx2.cpp.o] Error 1
make[4]: *** [CMakeFiles/Makefile2:480: kernels/CMakeFiles/embree_avx2.dir/all] Error 2
make[3]: *** [Makefile:156: all] Error 2
make[2]: *** [CMakeFiles/ext_embree.dir/build.make:86: embree/src/ext_embree-stamp/ext_embree-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:1743: CMakeFiles/ext_embree.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```


### Open3D, Python and System information

```markdown
- Operating system: Debian version: 11 (bullseye)
- Python version: Python 3.9 
- Open3D version: 0.16
- System architecture: rpi
- Is this a remote workstation?: no
- How did you install Open3D?: build from source
- Compiler version (if built from source): gcc 10.0
```


### Additional information

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,7 @@ endif()
 # CMAKE_HOST_SYSTEM_PROCESSOR is only available after calling project(),
 # which depends on ${OPEN3D_VERSION}, which depends on ${DEVELOPER_BUILD}.
 if(UNIX AND NOT APPLE)
-    execute_process(COMMAND uname -p
+    execute_process(COMMAND uname -m
         OUTPUT_VARIABLE PROCESSOR_ARCH
         OUTPUT_STRIP_TRAILING_WHITESPACE
     )

